### ISSUE TITLE
Cannot install Gazebo8 Mac OSX.

### ISSUE THREAD

**Issue by alex-born** [osrf/homebrew-simulation#323]
I tried to install gazebo8 on OSX (10.12.6) via the install script.

**Error from install script:**
```
/usr/local/Cellar/gazebo8/8.1.1_7/bin/gz
sdf


[libprotobuf ERROR google/protobuf/descriptor_database.cc:58] File already exists in database: ignition/msgs/altimeter.proto
[libprotobuf FATAL google/protobuf/descriptor.cc:1394] CHECK failed: generated_database_->Add(encoded_file_descriptor, size):
libc++abi.dylib: terminating with uncaught exception of type google::protobuf::FatalException: CHECK failed: generated_database_->Add(encoded_file_descriptor, size):
Error: osrf/simulation/gazebo8: failed
Failed executing: /usr/local/Cellar/gazebo8/8.1.1_7/bin/gz sdf
/usr/local/Homebrew/Library/Homebrew/formula.rb:1796:in `block in system'
/usr/local/Homebrew/Library/Homebrew/formula.rb:1734:in `open'
/usr/local/Homebrew/Library/Homebrew/formula.rb:1734:in `system'
/usr/local/Homebrew/Library/Taps/osrf/homebrew-simulation/gazebo8.rb:76:in `block in <class:Gazebo8>'
/usr/local/Homebrew/Library/Homebrew/formula.rb:1640:in `block (3 levels) in run_test'
/usr/local/Homebrew/Library/Homebrew/utils.rb:554:in `with_env'
/usr/local/Homebrew/Library/Homebrew/formula.rb:1639:in `block (2 levels) in run_test'
/usr/local/Homebrew/Library/Homebrew/formula.rb:834:in `with_logging'
/usr/local/Homebrew/Library/Homebrew/formula.rb:1638:in `block in run_test'
/usr/local/Homebrew/Library/Homebrew/extend/fileutils.rb:14:in `block in mktemp'
/usr/local/Homebrew/Library/Homebrew/extend/fileutils.rb:74:in `block in run'
/usr/local/Homebrew/Library/Homebrew/extend/fileutils.rb:74:in `chdir'
/usr/local/Homebrew/Library/Homebrew/extend/fileutils.rb:74:in `run'
/usr/local/Homebrew/Library/Homebrew/extend/fileutils.rb:13:in `mktemp'
/usr/local/Homebrew/Library/Homebrew/formula.rb:1632:in `run_test'
/usr/local/Homebrew/Library/Homebrew/test.rb:28:in `block in <main>'
/usr/local/Homebrew/Library/Homebrew/vendor/portable-ruby/2.3.3/lib/ruby/2.3.0/timeout.rb:91:in `block in timeout'
/usr/local/Homebrew/Library/Homebrew/vendor/portable-ruby/2.3.3/lib/ruby/2.3.0/timeout.rb:33:in `block in catch'
/usr/local/Homebrew/Library/Homebrew/vendor/portable-ruby/2.3.3/lib/ruby/2.3.0/timeout.rb:33:in `catch'
/usr/local/Homebrew/Library/Homebrew/vendor/portable-ruby/2.3.3/lib/ruby/2.3.0/timeout.rb:33:in `catch'
/usr/local/Homebrew/Library/Homebrew/vendor/portable-ruby/2.3.3/lib/ruby/2.3.0/timeout.rb:106:in `timeout'
/usr/local/Homebrew/Library/Homebrew/test.rb:27:in `<main>'
```
**I tried brew "upgrade `brew deps gazebo8`"**
```
Error: boost 1.65.1 already installed
Error: bullet 2.87 already installed
Error: cmake 3.10.0 already installed
Error: osrf/simulation/cppzmq 0.0.0-brew already installed
Error: doxygen 1.8.13_1 already installed
Error: freeimage 3.17.0_1 already installed
Error: freetype 2.8.1 already installed
Error: gettext 0.19.8.1 already installed
Error: glib 2.54.2 already installed
Error: gts 0.7.6 already installed
Error: osrf/simulation/ignition-math3 3.3.0 already installed
Error: osrf/simulation/ignition-msgs0 0.7.0_5 already installed
Error: osrf/simulation/ignition-tools 0.0.0-20160209-381c0a968fa already installed
Error: osrf/simulation/ignition-transport3 3.0.1_8 already installed
Error: jasper 2.0.14 already installed
Error: jpeg 9b already installed
Error: libffi 3.2.1 already installed
Error: libpng 1.6.34 already installed
Error: libtar 1.2.20 already installed
Error: libtiff 4.0.8_5 already installed
Error: libzzip 0.13.67 already installed
Error: netpbm 10.73.17 already installed
Error: osrf/simulation/ogre1.9 1.9-20160714-108ab0bcc69603dba32c0ffd4bbbc39051f421c9_4 already installed
Error: ossp-uuid 1.6.2_2 already installed
Error: pcre 8.41 already installed
Error: pkg-config 0.29.2 already installed
Error: protobuf 3.5.0 already installed
Error: protobuf-c 1.3.0_2 already installed
Error: qt 5.9.3 already installed
Error: qwt 6.1.3_4 already installed
Error: osrf/simulation/sdformat5 5.3.0 already installed
Error: osrf/simulation/simbody 3.5.4_2 already installed
Error: tbb 2018_U1 already installed
Error: tinyxml 2.6.2 already installed
Error: tinyxml2 5.0.1 already installed
Error: zeromq 4.2.2 already installed
```
**"brew doctor" Output:**
```
Please note that these warnings are just used to help the Homebrew maintainers
with debugging if you file an issue. If everything you use Homebrew for is
working fine: please don't worry and just ignore them. Thanks!

Warning: "config" scripts exist outside your system or Homebrew directories.
`./configure` scripts often look for *-config scripts to determine if
software packages are installed, and what additional flags to use when
compiling and linking.

Having additional scripts in your path can confuse software installed via
Homebrew if the config script overrides a system or Homebrew provided
script of the same name. We found the following "config" scripts:
  /opt/ImageMagick/bin/Magick++-config
  /opt/ImageMagick/bin/Magick-config
  /opt/ImageMagick/bin/MagickCore-config
  /opt/ImageMagick/bin/MagickWand-config
  /opt/ImageMagick/bin/Wand-config

Warning: Python is installed at /Library/Frameworks/Python.framework

Homebrew only supports building against the System-provided Python or a
brewed Python. In particular, Pythons installed to /Library can interfere
with other software installs.

Warning: Unbrewed dylibs were found in /usr/local/lib.
If you didn't put them there on purpose they could cause problems when
building Homebrew formulae, and may need to be deleted.

Unexpected dylibs:
  /usr/local/lib/libtcl8.6.dylib
  /usr/local/lib/libtk8.6.dylib

Warning: Unbrewed header files were found in /usr/local/include.
If you didn't put them there on purpose they could cause problems when
building Homebrew formulae, and may need to be deleted.

Unexpected header files:
  /usr/local/include/fakemysql.h
  /usr/local/include/fakepq.h
  /usr/local/include/fakesql.h
  /usr/local/include/gdb/jit-reader.h
  /usr/local/include/itcl.h
  /usr/local/include/itcl2TclOO.h
  /usr/local/include/itclDecls.h
  /usr/local/include/itclInt.h
  /usr/local/include/itclIntDecls.h
  /usr/local/include/itclMigrate2TclCore.h
  /usr/local/include/itclTclIntStubsFcn.h
  /usr/local/include/mysqlStubs.h
  /usr/local/include/node/android-ifaddrs.h
  /usr/local/include/node/ares.h
  /usr/local/include/node/ares_version.h
  /usr/local/include/node/libplatform/libplatform.h
  /usr/local/include/node/nameser.h
  /usr/local/include/node/node.h
  /usr/local/include/node/node_buffer.h
  /usr/local/include/node/node_internals.h
--- More Unexpected Header Files 
  /usr/local/include/tk.h
  /usr/local/include/tkDecls.h
  /usr/local/include/tkPlatDecls.h

Warning: Unbrewed .pc files were found in /usr/local/lib/pkgconfig.
If you didn't put them there on purpose they could cause problems when
building Homebrew formulae, and may need to be deleted.

Unexpected .pc files:
  /usr/local/lib/pkgconfig/tcl.pc
  /usr/local/lib/pkgconfig/tk.pc

Warning: Unbrewed static libraries were found in /usr/local/lib.
If you didn't put them there on purpose they could cause problems when
building Homebrew formulae, and may need to be deleted.

Unexpected static libraries:
  /usr/local/lib/libarm-frc-linux-gnueabi-sim.a
  /usr/local/lib/libtclstub8.6.a
  /usr/local/lib/libtkstub8.6.a

Warning: You have unlinked kegs in your Cellar
Leaving kegs unlinked can lead to build-trouble and cause brews that depend on
those kegs to fail to run properly once built. Run `brew link` on these:
  mupdf-tools
  ogre
  python
  python3
  qt@4
  qwt-qt4
  ruby
  sdformat5

Warning: Homebrew's sbin was not found in your PATH but you have installed
formulae that put executables in /usr/local/sbin.
Consider setting the PATH for example like so
  echo 'export PATH="/usr/local/sbin:$PATH"' >> ~/.zshrc

Warning: Some keg-only formula are linked into the Cellar.
Linking a keg-only formula, such as gettext, into the cellar with
`brew link <formula>` will cause other formulae to detect them during
the `./configure` step. This may cause problems when compiling those
other formulae.

Binaries provided by keg-only formulae may override system binaries
with other strange results.

You may wish to `brew unlink` these brews:
  qt
```
**"which gazebo" output:**
`
/usr/local/bin/gazebo
`
**"brew test gazebo8" output:**
```
Testing osrf/simulation/gazebo8
==> /usr/local/Cellar/gazebo8/8.1.1_7/bin/gz sdf
Last 15 lines from /Users/alexnrob/Library/Logs/Homebrew/gazebo8/test.01.gz:
2017-12-03 11:23:29 -0800

/usr/local/Cellar/gazebo8/8.1.1_7/bin/gz
sdf

[libprotobuf ERROR google/protobuf/descriptor_database.cc:58] File already exists in database: ignition/msgs/altimeter.proto
[libprotobuf FATAL google/protobuf/descriptor.cc:1394] CHECK failed: generated_database_->Add(encoded_file_descriptor, size):
libc++abi.dylib: terminating with uncaught exception of type google::protobuf::FatalException: CHECK failed: generated_database_->Add(encoded_file_descriptor, size):
Error: osrf/simulation/gazebo8: failed
Failed executing: /usr/local/Cellar/gazebo8/8.1.1_7/bin/gz sdf
/usr/local/Homebrew/Library/Homebrew/formula.rb:1796:in `block in system'
/usr/local/Homebrew/Library/Homebrew/formula.rb:1734:in `open'
/usr/local/Homebrew/Library/Homebrew/formula.rb:1734:in `system'
/usr/local/Homebrew/Library/Taps/osrf/homebrew-simulation/gazebo8.rb:76:in `block in <class:Gazebo8>'
/usr/local/Homebrew/Library/Homebrew/formula.rb:1640:in `block (3 levels) in run_test'
/usr/local/Homebrew/Library/Homebrew/utils.rb:554:in `with_env'
/usr/local/Homebrew/Library/Homebrew/formula.rb:1639:in `block (2 levels) in run_test'
/usr/local/Homebrew/Library/Homebrew/formula.rb:834:in `with_logging'
/usr/local/Homebrew/Library/Homebrew/formula.rb:1638:in `block in run_test'
/usr/local/Homebrew/Library/Homebrew/extend/fileutils.rb:14:in `block in mktemp'
/usr/local/Homebrew/Library/Homebrew/extend/fileutils.rb:74:in `block in run'
/usr/local/Homebrew/Library/Homebrew/extend/fileutils.rb:74:in `chdir'
/usr/local/Homebrew/Library/Homebrew/extend/fileutils.rb:74:in `run'
/usr/local/Homebrew/Library/Homebrew/extend/fileutils.rb:13:in `mktemp'
/usr/local/Homebrew/Library/Homebrew/formula.rb:1632:in `run_test'
/usr/local/Homebrew/Library/Homebrew/test.rb:28:in `block in <main>'
/usr/local/Homebrew/Library/Homebrew/vendor/portable-ruby/2.3.3/lib/ruby/2.3.0/timeout.rb:91:in `block in timeout'
/usr/local/Homebrew/Library/Homebrew/vendor/portable-ruby/2.3.3/lib/ruby/2.3.0/timeout.rb:33:in `block in catch'
/usr/local/Homebrew/Library/Homebrew/vendor/portable-ruby/2.3.3/lib/ruby/2.3.0/timeout.rb:33:in `catch'
/usr/local/Homebrew/Library/Homebrew/vendor/portable-ruby/2.3.3/lib/ruby/2.3.0/timeout.rb:33:in `catch'
/usr/local/Homebrew/Library/Homebrew/vendor/portable-ruby/2.3.3/lib/ruby/2.3.0/timeout.rb:106:in `timeout'
/usr/local/Homebrew/Library/Homebrew/test.rb:27:in `<main>'
```
**"brew upgrade protobuf":**
```
Error: protobuf 3.5.0 already installed
```
**" brew upgrade ignition-msgs0":**
```
Error: osrf/simulation/ignition-msgs0 0.7.0_5 already installed
```

**Comment by scpeters** [comment]
I can't reproduce the problem. I just installed the `gazebo8` bottle on our 10.12 machine and it was able to run the test successfully.

I have seen the "File already exists in database" protobuf errors before, and I don't remember exactly what happened, but I think my environment and/or environment variables were at fault.

**Comment by scpeters** [comment]
I just remembered one way to reproduce the "File already exists in database" protobuf error. If you compile gazebo but have a different set of gazebo libraries earlier in your `DYLD_LIBRARY_PATH`, then you might see this error. The key is making sure the correct libraries are loaded.

For example, I just noticed this error after `brew install gazebo8`, then compiling gazebo from source without installing, and running a test. After `brew unlink gazebo8`, I was able to run the test successfully.
